Hank-iOS/ChatViewModel.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ChatViewModel.swift
+//  Hank-iOS
+//
+//  Created by Craig Atallah Frost on 26/06/2024.
+//
+
+import SwiftUI
+
+@Observable @MainActor
+class ChatViewModel {
+    var messages: [Message] = []
+    var currentInput: String = ""
+    var isLoading = false
+    
+    private let anthropicService = AnthropicService()
+    
+    func sendMessage() {
+        guard !currentInput.isEmpty else { return }
+        
+        let userMessage = Message(role: "user", content: currentInput)
+        
+        DispatchQueue.main.async {
+            self.messages.append(userMessage)
+            self.currentInput = ""
+            self.isLoading = true
+        }
+        
+        Task {
+            do {
+                let response = try await anthropicService.sendMessages(self.messages)
+                DispatchQueue.main.async {
+                    let assistantMessage = Message(role: "assistant", content: response)
+                    self.messages.append(assistantMessage)
+                    self.isLoading = false
+                }
+            } catch {
+                DispatchQueue.main.async {
+                    print("Error: \(error.localizedDescription)")
```

Comment:
It might be good to consider adding an user alert for receiving errors.

Reply:
Yeah definitely 👍 mostly missing for the same reason as comment above: hadn't yet figured out the best way to present errors to the user. Do folks more commonly use dialogs? Toasts? Some sort of inline error/banner etc.?

Reply:
Specifically asking about iOS conventions here and whether you have any context or opinions. I'm used to building stuff for the web and for larger screens, so super interested in hearing from folks who work with mobile.